base/math.jl

```diff
@@ -1292,7 +1292,7 @@ end
     return copysign(pow_body(abs(x), y), s)
 end
 
-@assume_effects :foldable @noinline function pow_body(x::Float64, y::Float64)
+@assume_effects :foldable @inline function pow_body(x::Float64, y::Float64)
```

Comment:
I introduced this, because I wanted to calm the last test in `compiler/codegen.jl`, which was introduced for PR #52194. In v1.10.2, this method is also inlined, and `exp_impl` is @inline at the definition site.
That did not work, because I could not enforce `exp_impl` to be inlined into `^` this way.

I would prefer to change that compiler test to make the implementation of `^` independent of those obscure multiversioning issues. @gbaraldi

Reply:
yeah, I think you could change the test to just be something like `evalpoly(Float64, NTuple{10, Float64})` rather than `^(Float64,Float64)`.